modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/service/impl/KBArticleServiceImpl.java

```diff
@@ -281,6 +282,25 @@ public KBArticle fetchLatestKBArticleByUrlTitle(
 		return null;
 	}
 
+	@Override
+	public Lock forceLockKBArticle(long groupId, long resourcePrimKey)
+		throws PortalException {
+
+		PermissionChecker permissionChecker = getPermissionChecker();
+
+		if (!permissionChecker.isGroupAdmin(groupId)) {
+			throw new PrincipalException.MustBeCompanyAdmin(permissionChecker);
```

Comment:
I understand the logic but semantics seems wrong.

A groupAdmin is not a companyAdmin. Do we have other choices? Maybe we should have to add a new "MustBeGroupAdmin" in PrincipalException class.

Reply:
I saw the same usage in `SaveGroupConfigurationMVCActionCommand`, but I agree with you, it is better to add the "MustBeGroupAdmin" exception.

Reply:
Change done here [095fffa](https://github.com/liferay-content-management/liferay-portal/pull/5187/commits/095fffa14a3d36eceebd9ed200241b75c0886db3)

See exception message: I'm using "site administrator" instead of "group administrator". That's right isn't?

Reply:
Technically, a group could be an asset library, but as we don't support KB in asset libraries it is fine as is.